modules/pel/peltool/user_data.py

```diff
@@ -37,6 +37,10 @@ def toJSON(self) -> OrderedDict:
 
         value = parser.parse()
 
+        # Catch if value is None, otherwise python crashes
+        if value is None:
```

Comment:
A few things:

The ExtendedUserData section also parses user data, that's why the ParseUserData class up above is used, so the code that interacts with the plugins is just in one place.  So any changes would be done in that class.

Also the changes here wouldn't even print a hex dump of the data, which I think would be better than nothing.

I think you could do something similar to this error path that is already there:

            print('Failed parsing user data for creator {} compID {} '
                  'subType {} version {}: {}'.format(
                      self.creatorID, "0x%04X" % self.compID,
                      "0x%X" % self.subType, self.version, str(e)), file=sys.stderr)
            mv = memoryview(self.data)
            return json.dumps(hexdump(mv))

if you wanted, I suppose you could add the error into the data along with the hexdump.  'plugin returned None' or something like that being the message.

Reply:
Thanks, I'll setup a recreate of the error paths and see how this can be modified.